radis/lbl/base.py

```diff
@@ -1497,6 +1521,129 @@ def _calc_noneq_parameters(self, vib_distribution, singleTvibmode):
             else:
                 required_columns = ["Evib1l", "Evib2l", "Evib3l"]
 
+        if not all_in(required_columns, df):
+            # Checking if the cache file exists or not, retrieving data
+            # if it's present else regenerating the files.
+            if exists(cache_filename):
+                existing_file_metadata = DataFileManager(engine).read_metadata(
+                    cache_filename
+                )
+                cache_file_data = DataFileManager(engine).read(cache_filename)
+                # Checking whether the right columns and indices are present in the cache file
+                try:
+                    # Checking if the same isotopes are present in the cache file as required
+                    if existing_file_metadata["isotope"] == isotope:
+                        # assert existing_file_metadata["number_of_lines"] >= (df.index[-1] - df.index[0])
+
+                        # Checking if the dataframe indices are present in the correct
+                        # range, otherwise regenerating cache file
+                        if existing_file_metadata["df_last_index"] == df.index[-1]:
+                            # Checking that all cache file metadata is correct:
+                            if "number_of_lines" in existing_file_metadata:
```

Comment:
If cache file didn't have metadata key, don't raise an error : just regenerate the cache file.

- replace `(if value == metadata[key] then... Else:regenerate)`  statements by  `(if key in metadata and value == metadata[key] then... Else:regenerate)` (they will check that Key exist before reading its value. If it doesn't exist Python stop the evaluation before the "and" therefore there won't be any KeyError

- add a test and add assert statements there, to check thar metadata exists